Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 3226)
+++ CMakeLists.txt	(revision 3227)
@@ -25,6 +25,18 @@
     add_definitions(-D_SECURE_SCL=0)
 endif(MSVC)
 
+# try to find system copy of qtsingleapplication
+find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
+find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
+if(QTSINGLEAPPLICATION_INCLUDE_DIRS AND QTSINGLEAPPLICATION_LIBRARIES)
+    message(STATUS "Found QtSingleApplication: ${QTSINGLEAPPLICATION_LIBRARIES}")
+else()
+    message(STATUS "QtSingleApplication not found; using bundled copy")
+    add_subdirectory(qtsingleapplication)
+    set(QTSINGLEAPPLICATION_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/qtsingleapplication/src)
+    set(QTSINGLEAPPLICATION_LIBRARIES qtsingleapplication)
+endif()
+
 set(QT_USE_QTNETWORK true)
 set(QT_USE_QTWEBKIT true)
 set(QT_USE_QTXML true)
@@ -99,8 +111,6 @@
 
 include(${QT_USE_FILE})
 
-add_subdirectory( qtsingleapplication )
-
 find_package(Subversion)
 if (Subversion_FOUND AND EXISTS ${PROJECT_SOURCE_DIR}/.svn)
   Subversion_WC_INFO(${PROJECT_SOURCE_DIR} PROJECT)
@@ -113,10 +123,10 @@
 include_directories(
     ${CMAKE_SOURCE_DIR}
     ${CMAKE_BINARY_DIR}
+    ${LIBP11_INCLUDE_DIR}
     ${OPENSSL_INCLUDE_DIR}
-    ${LIBP11_INCLUDE_DIR}
+    ${QTSINGLEAPPLICATION_INCLUDE_DIRS}
     ${SMARTCARDPP_INCLUDE_DIR}
-    qtsingleapplication/src
 )
 
 if(APPLE)
@@ -170,13 +180,13 @@
 target_link_libraries(qesteidutil
     ${QT_QTMAIN_LIBRARY}
     ${QT_LIBRARIES}
+    ${LIBP11_LIBRARIES}
     ${OPENSSL_LIBRARIES}
     ${OPENSSLCRYPTO_LIBRARIES}
-    ${LIBP11_LIBRARIES}
+    ${QTSINGLEAPPLICATION_LIBRARIES}
     ${SMARTCARDPP_LIBRARIES}
-	${WIN_LIBRARIES}
+    ${WIN_LIBRARIES}
     ${CARBON_LIBRARY}
-    qtsingleapplication
 )
 
 if(UNIX AND NOT APPLE)
