--- qesteidutil-0.3.1/CMakeLists.txt~	2011-07-06 19:50:13.000000000 +0300
+++ qesteidutil-0.3.1/CMakeLists.txt	2012-01-08 13:51:46.645405478 +0200
@@ -35,7 +35,7 @@
 include(${QT_USE_FILE})
 
 # try to find system copy of qtsingleapplication
-find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
+find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES qt4/QtSolutions QtSolutions)
 find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
 if(QTSINGLEAPPLICATION_INCLUDE_DIRS AND QTSINGLEAPPLICATION_LIBRARIES)
     message(STATUS "Found QtSingleApplication: ${QTSINGLEAPPLICATION_LIBRARIES}")
