--- qesteidutil-3.12.10/common/CMakeLists.txt~	2017-11-22 14:34:35.000000000 +0200
+++ qesteidutil-3.12.10/common/CMakeLists.txt	2018-08-22 08:52:33.867141765 +0300
@@ -44,8 +44,7 @@
 	set( LIBRARIES "-framework Cocoa -framework PCSC -fobjc-arc" )
 else()
 	include_directories( ${CMAKE_SOURCE_DIR} )
-	add_subdirectory( qtsingleapplication )
-	list( APPEND LIBRARIES qtsingleapplication )
+	list( APPEND LIBRARIES QtSolutions_SingleApplication-2.6)
 endif()
 
 configure_file( translations/common_tr.qrc common_tr.qrc COPYONLY )
--- qesteidutil-3.8.0.1106/common/Common.h~	2013-12-02 11:48:37.000000000 +0200
+++ qesteidutil-3.8.0.1106/common/Common.h	2015-02-22 20:09:05.556946877 +0200
@@ -32,7 +32,7 @@
 #endif
 typedef QApplication BaseApplication;
 #else
-#include "qtsingleapplication/src/QtSingleApplication"
+#include "QtSolutions/QtSingleApplication"
 typedef QtSingleApplication BaseApplication;
 #endif
 
--- qesteidutil-3.8.0.1106/src/main.cpp~	2015-02-23 09:19:12.000000000 +0200
+++ qesteidutil-3.8.0.1106/src/main.cpp	2015-02-23 09:19:35.942253525 +0200
@@ -63,7 +63,8 @@
 #ifndef Q_OS_MAC
 	if( app.isRunning() )
 	{
-		app.sendMessage( "" );
+		QString message = "";
+		app.sendMessage( message );
 		return 0;
 	}
 #endif
